src/lib/gameService.ts

```diff
@@ -12,41 +13,172 @@ export class GameService {
 			code: this.generateCode(),
 			currentPlayer: undefined,
 			players: [],
-			currentBid: undefined
+			currentBid: undefined,
+			maxPlayers: 8,
+			numberOfPlayers: 0,
+			initialDiceCount: 6
 		};
 
 		this.repository.saveGame(game);
 
 		return game.code;
 	}
 
-	public async addPlayer(name: string, gameCode: string): Promise<string> {
-		throw new Error('not implemented');
+	public async addPlayer(playerName: string, gameCode: string): Promise<string> {
+		const game: Game | undefined = await gameRepository.getGame(gameCode);
+
+		if (!game) {
+			throw new Error('Game is undefined');
+		}
+
+		if (game.state === GameState.InProgress) {
+			throw new Error('The game is already in progress');
+		} else if (game.state === GameState.Finished) {
+			throw new Error('The game has already finished');
+		}
+
+		const newPlayer: Player = {
+			name: playerName,
+			code: gameCode,
```

Comment:
I didn't explain my thinking earlier, but the `playerCode` will be another random string unique to the player. Then each player has a unique `playerToken` that looks like `gameCode-playerCode` that can be used for a light form of authorisation.

The player code can't be the player name because then other players could impersonate the player if they know the player's name.